### ISSUE TITLE
Error between kafka-redis-sink-l4t and kafka-redis-adapter-x86

### ISSUE THREAD

**Issue by ayiskakov** [insight-platform/Savant#671]
An error occurs when attempting to deserialize messages in the kafka-redis-source-1 container, specifically when processing video frames. The system fails to deserialize the message due to an InvalidTag(37) error within a VideoFrame payload, leading to a runtime error and stopping the deserializer, sender, and poller processes.

### Steps to Reproduce:
1. Set up the Docker environment as described in the provided Docker Compose files for both L4T and x86 architectures.
2. Start the Docker containers using the provided Docker Compose configurations.
3. Observe the error in the kafka-redis-source-1 container logs after attempting to process messages.

### Expected Behavior:
The system should correctly deserialize messages received from Kafka and process them without errors.

### Actual Behavior:
The deserialization process fails with a runtime error due to an Unknown message containing an InvalidTag(138) within the payload of a VideoFrame. This error halts further processing of messages, stopping the deserializer, sender, and poller processes.

logs
```text
kafka-redis-source-1  | stty: 'standard input': Inappropriate ioctl for device
kafka-redis-source-1  |  ERROR insight::savant::adapters::kafka_redis::KafkaRedisSource > Failed to deserialize message: Unknown message: Message { meta: MessageMeta { lib_version: "0.1.85", routing_labels: [], span_context: PropagatedContext({}), seq_id: 0 }, payload: Unknown("CheckBytesError(CheckBytesError(StructCheckError { field_name: \"payload\", inner: InvalidTuple { variant_name: \"VideoFrame\", inner: TupleStructCheckError { field_index: 0, inner: ValueCheckBytesError(StructCheckError { field_name: \"content\", inner: InvalidTag(37) }) } } }))") }
kafka-redis-source-1  | Aliases for entries in sys.path:
kafka-redis-source-1  |     <pwd>: /opt/savant
kafka-redis-source-1  | Traceback (most recent call last):
kafka-redis-source-1  |     <pwd> /adapters/python/sources/kafka_redis.py:141  messages_processor  deserialized = await self.process_message(data)
kafka-redis-source-1  |     <pwd> /adapters/python/sources/kafka_redis.py:192  process_message     raise RuntimeError(f'Unknown message: {message}')
kafka-redis-source-1  | RuntimeError: Unknown message: Message { meta: MessageMeta { lib_version: "0.1.85", routing_labels: [], span_context: PropagatedContext({}), seq_id: 0 }, payload: Unknown("CheckBytesError(CheckBytesError(StructCheckError { field_name: \"payload\", inner: InvalidTuple { variant_name: \"VideoFrame\", inner: TupleStructCheckError { field_index: 0, inner: ValueCheckBytesError(StructCheckError { field_name: \"content\", inner: InvalidTag(37) }) } } }))") }
```

**Comment by ayiskakov** [comment]
`
version: "3.3"
services:
  kafka:
    container_name: kafka
    image: bitnami/kafka:3.5.1
    restart: unless-stopped
    # volumes:
    #   - "kafka_data:/bitnami"
    environment:
      # KRaft settings
      - KAFKA_CFG_NODE_ID=0
      - KAFKA_CFG_PROCESS_ROLES=controller,broker
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
      # Listeners
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://172.20.10.12:9092
      # - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT
    ports:
      - "9092:9092"
    healthcheck:
      test:
        - CMD-SHELL
        - /opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9092 --list
      start_period: 60s
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - centrifugo

  keydb:
    image: eqalpha/keydb:x86_64_v6.3.3
    restart: unless-stopped
    volumes:
      - keydb_data:/data
    ports:
      - "6379:6379"
    networks:
      - centrifugo

  kafka-redis-source:
    image: ghcr.io/insight-platform/savant-adapters-py:0.2.9
    restart: unless-stopped
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
    environment:
      - ZMQ_ENDPOINT=pub+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - KAFKA_BROKERS=kafka:9092
      - KAFKA_TOPIC=kafka-redis-adapter-demo
      - KAFKA_GROUP_ID=kafka-redis-adapter-demo
      - KAFKA_CREATE_TOPIC=True
      - KAFKA_CREATE_TOPIC_NUM_PARTITIONS=4
      - KAFKA_CREATE_TOPIC_REPLICATION_FACTOR=1
      - 'KAFKA_CREATE_TOPIC_CONFIG={"retention.ms": 300000}'
    command: python -m adapters.python.sources.kafka_redis
    depends_on:
      kafka:
        condition: service_healthy
      keydb:
        condition: service_started
    networks:
      - centrifugo
 
  module:
    build:
      context: .
      dockerfile: docker/Dockerfile.x86
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
      - ./tmp:/tmp
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
    ports:
      - "8888:8888"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    networks:
      - network
      - centrifugo

  always-on-sink:
    image: ghcr.io/insight-platform/savant-adapters-deepstream:0.2.9-6.3
    restart: unless-stopped
    ports:
      - "554:554"    # RTSP
      - "1935:1935"  # RTMP
      - "888:888"    # HLS
      - "8889:8889"  # WebRTC
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./assets/stub_imgs:/stub_imgs
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - SOURCE_ID=video
      - STUB_FILE_LOCATION=/stub_imgs/smpte100_1280x720.jpeg
      - DEV_MODE=True
      - FRAMERATE=25/1
    command: python -m adapters.ds.sinks.always_on_rtsp
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]

  kafka-ui:
    container_name: kafka-ui
    image: provectuslabs/kafka-ui:latest
    ports:
      - 8082:8080
    environment:
      DYNAMIC_CONFIG_ENABLED: true
    networks:
      - network
      - centrifugo

volumes:
  zmq_sockets_server:
  # kafka_data:
  keydb_data:

networks:
  network:
    attachable: true
  centrifugo:
    external: true

`

**Comment by ayiskakov** [comment]
`version: "3.3"
services:
  kafka:
    container_name: kafka
    image: bitnami/kafka:3.5.1
    restart: unless-stopped
    # volumes:
    #   - "kafka_data:/bitnami"
    environment:
      # KRaft settings
      - KAFKA_CFG_NODE_ID=0
      - KAFKA_CFG_PROCESS_ROLES=controller,broker
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
      # Listeners
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://172.20.10.12:9092
      # - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT
    ports:
      - "9092:9092"
    healthcheck:
      test:
        - CMD-SHELL
        - /opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9092 --list
      start_period: 60s
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - centrifugo

  keydb:
    image: eqalpha/keydb:x86_64_v6.3.3
    restart: unless-stopped
    volumes:
      - keydb_data:/data
    ports:
      - "6379:6379"
    networks:
      - centrifugo

  kafka-redis-source:
    image: ghcr.io/insight-platform/savant-adapters-py:0.2.9
    restart: unless-stopped
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
    environment:
      - ZMQ_ENDPOINT=pub+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - KAFKA_BROKERS=kafka:9092
      - KAFKA_TOPIC=kafka-redis-adapter-demo
      - KAFKA_GROUP_ID=kafka-redis-adapter-demo
      - KAFKA_CREATE_TOPIC=True
      - KAFKA_CREATE_TOPIC_NUM_PARTITIONS=4
      - KAFKA_CREATE_TOPIC_REPLICATION_FACTOR=1
      - 'KAFKA_CREATE_TOPIC_CONFIG={"retention.ms": 300000}'
    command: python -m adapters.python.sources.kafka_redis
    depends_on:
      kafka:
        condition: service_healthy
      keydb:
        condition: service_started
    networks:
      - centrifugo
 
  module:
    build:
      context: .
      dockerfile: docker/Dockerfile.x86
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
      - ./tmp:/tmp
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
    ports:
      - "8888:8888"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    networks:
      - network
      - centrifugo

  always-on-sink:
    image: ghcr.io/insight-platform/savant-adapters-deepstream:0.2.9-6.3
    restart: unless-stopped
    ports:
      - "554:554"    # RTSP
      - "1935:1935"  # RTMP
      - "888:888"    # HLS
      - "8889:8889"  # WebRTC
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./assets/stub_imgs:/stub_imgs
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - SOURCE_ID=video
      - STUB_FILE_LOCATION=/stub_imgs/smpte100_1280x720.jpeg
      - DEV_MODE=True
      - FRAMERATE=25/1
    command: python -m adapters.ds.sinks.always_on_rtsp
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]

  kafka-ui:
    container_name: kafka-ui
    image: provectuslabs/kafka-ui:latest
    ports:
      - 8082:8080
    environment:
      DYNAMIC_CONFIG_ENABLED: true
    networks:
      - network
      - centrifugo

volumes:
  zmq_sockets_server:
  # kafka_data:
  keydb_data:

networks:
  network:
    attachable: true
  centrifugo:
    external: true
`

**Comment by ayiskakov** [comment]
`yaml
version: "3.3"
services:
  kafka:
    container_name: kafka
    image: bitnami/kafka:3.5.1
    restart: unless-stopped
    # volumes:
    #   - "kafka_data:/bitnami"
    environment:
      # KRaft settings
      - KAFKA_CFG_NODE_ID=0
      - KAFKA_CFG_PROCESS_ROLES=controller,broker
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
      # Listeners
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://172.20.10.12:9092
      # - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT
    ports:
      - "9092:9092"
    healthcheck:
      test:
        - CMD-SHELL
        - /opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9092 --list
      start_period: 60s
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - centrifugo

  keydb:
    image: eqalpha/keydb:x86_64_v6.3.3
    restart: unless-stopped
    volumes:
      - keydb_data:/data
    ports:
      - "6379:6379"
    networks:
      - centrifugo

  kafka-redis-source:
    image: ghcr.io/insight-platform/savant-adapters-py:0.2.9
    restart: unless-stopped
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
    environment:
      - ZMQ_ENDPOINT=pub+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - KAFKA_BROKERS=kafka:9092
      - KAFKA_TOPIC=kafka-redis-adapter-demo
      - KAFKA_GROUP_ID=kafka-redis-adapter-demo
      - KAFKA_CREATE_TOPIC=True
      - KAFKA_CREATE_TOPIC_NUM_PARTITIONS=4
      - KAFKA_CREATE_TOPIC_REPLICATION_FACTOR=1
      - 'KAFKA_CREATE_TOPIC_CONFIG={"retention.ms": 300000}'
    command: python -m adapters.python.sources.kafka_redis
    depends_on:
      kafka:
        condition: service_healthy
      keydb:
        condition: service_started
    networks:
      - centrifugo
 
  module:
    build:
      context: .
      dockerfile: docker/Dockerfile.x86
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
      - ./tmp:/tmp
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
    ports:
      - "8888:8888"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    networks:
      - network
      - centrifugo

  always-on-sink:
    image: ghcr.io/insight-platform/savant-adapters-deepstream:0.2.9-6.3
    restart: unless-stopped
    ports:
      - "554:554"    # RTSP
      - "1935:1935"  # RTMP
      - "888:888"    # HLS
      - "8889:8889"  # WebRTC
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./assets/stub_imgs:/stub_imgs
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - SOURCE_ID=video
      - STUB_FILE_LOCATION=/stub_imgs/smpte100_1280x720.jpeg
      - DEV_MODE=True
      - FRAMERATE=25/1
    command: python -m adapters.ds.sinks.always_on_rtsp
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]

  kafka-ui:
    container_name: kafka-ui
    image: provectuslabs/kafka-ui:latest
    ports:
      - 8082:8080
    environment:
      DYNAMIC_CONFIG_ENABLED: true
    networks:
      - network
      - centrifugo

volumes:
  zmq_sockets_server:
  # kafka_data:
  keydb_data:

networks:
  network:
    attachable: true
  centrifugo:
    external: true

`

**Comment by ayiskakov** [comment]
docker-compose.x86.yml
```yml
version: "3.3"
services:

  kafka:
    container_name: kafka
    image: bitnami/kafka:3.5.1
    restart: unless-stopped
    # volumes:
    #   - "kafka_data:/bitnami"
    environment:
      # KRaft settings
      - KAFKA_CFG_NODE_ID=0
      - KAFKA_CFG_PROCESS_ROLES=controller,broker
      - KAFKA_CFG_CONTROLLER_QUORUM_VOTERS=0@kafka:9093
      # Listeners
      - KAFKA_CFG_LISTENERS=PLAINTEXT://:9092,CONTROLLER://:9093
      - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://172.20.10.12:9092
      # - KAFKA_CFG_ADVERTISED_LISTENERS=PLAINTEXT://:9092
      - KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CONTROLLER:PLAINTEXT,PLAINTEXT:PLAINTEXT
      - KAFKA_CFG_CONTROLLER_LISTENER_NAMES=CONTROLLER
      - KAFKA_CFG_INTER_BROKER_LISTENER_NAME=PLAINTEXT
    ports:
      - "9092:9092"
    healthcheck:
      test:
        - CMD-SHELL
        - /opt/bitnami/kafka/bin/kafka-topics.sh --bootstrap-server localhost:9092 --list
      start_period: 60s
      interval: 10s
      timeout: 5s
      retries: 5
    networks:
      - centrifugo

  keydb:
    image: eqalpha/keydb:x86_64_v6.3.3
    restart: unless-stopped
    volumes:
      - keydb_data:/data
    ports:
      - "6379:6379"
    networks:
      - centrifugo

  kafka-redis-source:
    image: ghcr.io/insight-platform/savant-adapters-py:0.2.9
    restart: unless-stopped
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
    environment:
      - ZMQ_ENDPOINT=pub+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - KAFKA_BROKERS=kafka:9092
      - KAFKA_TOPIC=kafka-redis-adapter-demo
      - KAFKA_GROUP_ID=kafka-redis-adapter-demo
      - KAFKA_CREATE_TOPIC=True
      - KAFKA_CREATE_TOPIC_NUM_PARTITIONS=4
      - KAFKA_CREATE_TOPIC_REPLICATION_FACTOR=1
      - 'KAFKA_CREATE_TOPIC_CONFIG={"retention.ms": 300000}'
      - REDIS_HOST=keydb
      - REDIS_PORT=6379
      - REDIS_DB=0
      - REDIS_TTL_SECONDS=60
    command: python -m adapters.python.sources.kafka_redis
    depends_on:
      kafka:
        condition: service_healthy
      keydb:
        condition: service_started
    networks:
      - centrifugo
 
  module:
    build:
      context: .
      dockerfile: docker/Dockerfile.x86
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
      - ./tmp:/tmp
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
    ports:
      - "8888:8888"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    networks:
      - network
      - centrifugo

  always-on-sink:
    image: ghcr.io/insight-platform/savant-adapters-deepstream:0.2.9-6.3
    restart: unless-stopped
    ports:
      - "554:554"    # RTSP
      - "1935:1935"  # RTMP
      - "888:888"    # HLS
      - "8889:8889"  # WebRTC
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./assets/stub_imgs:/stub_imgs
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - SOURCE_ID=video
      - STUB_FILE_LOCATION=/stub_imgs/smpte100_1280x720.jpeg
      - DEV_MODE=True
      - FRAMERATE=25/1
    command: python -m adapters.ds.sinks.always_on_rtsp
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]

  kafka-ui:
    container_name: kafka-ui
    image: provectuslabs/kafka-ui:latest
    ports:
      - 8082:8080
    environment:
      DYNAMIC_CONFIG_ENABLED: true
    networks:
      - network
      - centrifugo

volumes:
  zmq_sockets_server:
  # kafka_data:
  keydb_data:

networks:
  network:
    attachable: true
  centrifugo:
    external: true
```

**Comment by ayiskakov** [comment]
docker-compose.l4t.yml
```yml
version: "3.3"
services:

  usb-cam-source:
    image: ghcr.io/insight-platform/savant-adapters-gstreamer-l4t:0.2.9
    restart: unless-stopped
    devices: 
      - /dev/video0:/dev/video0
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      # - /dev/video0:/dev/video0
      - /tmp/video-loop-source-downloads:/tmp/video-loop-source-downloads
    environment:
      - DEVICE=/dev/video0
      - ZMQ_ENDPOINT=dealer+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - SOURCE_ID=video
      - SYNC_OUTPUT=True
      - FRAMERATE=30/1
    entrypoint: /opt/savant/adapters/gst/sources/usb_cam.sh
    depends_on:
      module:
        condition: service_healthy

  module:
    image: module-l4t
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=router+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
      - MODE=jetson
    ports:
      - "8888:8888"
    runtime: nvidia
    networks:
      - network

  kafka-redis-sink:
    image: ghcr.io/insight-platform/savant-adapters-py-l4t:0.2.9
    restart: unless-stopped
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - KAFKA_BROKERS=172.20.10.12:9092
      - KAFKA_TOPIC=kafka-redis-adapter-demo
      - KAFKA_CREATE_TOPIC=True
      - KAFKA_CREATE_TOPIC_NUM_PARTITIONS=4
      - KAFKA_CREATE_TOPIC_REPLICATION_FACTOR=1
      - 'KAFKA_CREATE_TOPIC_CONFIG={"retention.ms": 300000}'
      - REDIS_HOST=172.20.10.12
      - REDIS_PORT=6379
      - REDIS_DB=0
      - REDIS_TTL_SECONDS=60
    command: python -m adapters.python.sinks.kafka_redis

volumes:
  zmq_sockets:

networks:
  network:
    attachable: true

```

**Comment by ayiskakov** [comment]
## x86
```yml
version: "3.3"
services:

 
  module:
    build:
      context: .
      dockerfile: docker/Dockerfile.x86
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
      - ./tmp:/tmp
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=sub+bind:tcp://0.0.0.0:6666
      - ZMQ_SINK_ENDPOINT=pub+bind:ipc:///tmp/zmq-sockets/output-video.ipc
    ports:
      - "8888:8888"
      - "6666:6666"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    networks:
      - network
      - centrifugo

  always-on-sink:
    image: ghcr.io/insight-platform/savant-adapters-deepstream:0.2.9-6.3
    restart: unless-stopped
    ports:
      - "554:554"    # RTSP
      - "1935:1935"  # RTMP
      - "888:888"    # HLS
      - "8889:8889"  # WebRTC
    volumes:
      - zmq_sockets_server:/tmp/zmq-sockets
      - ./assets/stub_imgs:/stub_imgs
    environment:
      - ZMQ_ENDPOINT=sub+connect:ipc:///tmp/zmq-sockets/output-video.ipc
      - SOURCE_ID=video
      - STUB_FILE_LOCATION=/stub_imgs/smpte100_1280x720.jpeg
      - DEV_MODE=True
      - FRAMERATE=25/1
    command: python -m adapters.ds.sinks.always_on_rtsp
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
```

## l4t
```yml
version: "3.3"
services:

  usb-cam-source:
    image: ghcr.io/insight-platform/savant-adapters-gstreamer-l4t:0.2.9
    restart: unless-stopped
    devices: 
      - /dev/video0:/dev/video0
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      # - /dev/video0:/dev/video0
      - /tmp/video-loop-source-downloads:/tmp/video-loop-source-downloads
    environment:
      - DEVICE=/dev/video0
      - ZMQ_ENDPOINT=dealer+connect:ipc:///tmp/zmq-sockets/input-video.ipc
      - SOURCE_ID=video
      - SYNC_OUTPUT=True
      - FRAMERATE=30/1
    entrypoint: /opt/savant/adapters/gst/sources/usb_cam.sh
    depends_on:
      module:
        condition: service_healthy

  module:
    image: module-l4t
    volumes:
      - zmq_sockets:/tmp/zmq-sockets
      - ./models_cache:/models
      - ./downloads_cache:/downloads
      - ./src/module:/opt/savant/src/module
      - ./assets/ddnet:/models/ddnet
    environment:
      - HEALTHCHECK_PORT=8888
      - ZMQ_SRC_ENDPOINT=router+bind:ipc:///tmp/zmq-sockets/input-video.ipc
      - ZMQ_SINK_ENDPOINT=pub+connect:tcp://172.20.10.12:6666
      - MODE=jetson
    ports:
      - "8888:8888"
    runtime: nvidia
    networks:
      - network
```

**Comment by tomskikh** [comment]
I've managed to reproduce the issue on version `0.2.9`, but not on `latest`. @ayiskakov could you try to use images with tag `latest` and check if the bug is still there?
```
docker pull ghcr.io/insight-platform/savant-deepstream:latest
docker pull ghcr.io/insight-platform/savant-adapters-deepstream:latest
docker pull ghcr.io/insight-platform/savant-adapters-gstreamer:latest
docker pull ghcr.io/insight-platform/savant-adapters-py:latest

docker pull ghcr.io/insight-platform/savant-deepstream-l4t:latest
docker pull ghcr.io/insight-platform/savant-adapters-deepstream-l4t:latest
docker pull ghcr.io/insight-platform/savant-adapters-gstreamer-l4t:latest
docker pull ghcr.io/insight-platform/savant-adapters-py-l4t:latest
```

**Comment by bwsw** [comment]
Fixed in the 0.2.10
